src/helm/benchmark/presentation/summarize.py

```diff
@@ -315,6 +319,133 @@ def read_runs(self):
                 self.group_adapter_to_runs[group_name][adapter_spec].append(run)
                 self.group_scenario_adapter_to_runs[group_name][scenario_spec][adapter_spec].append(run)
 
+    def read_overlap_stats(self):
+        """
+        Load the overlap stats in the run suite path.
+        Concretely:
+            - get group -> scenario_spec information from self.runs
+                run_spec data
+            - read the files in the data_overlap directory in run_suit_path
+                which are scenario_spec -> overlap ids
+            - get aggregate stats for group -> overlap ratio
+        """
+
+        def get_group_to_scenario_specs(run_specs: List[RunSpec]) -> Dict[RunSpec, ScenarioSpec]:
+            scenario_specs_to_groups: Dict = dict()
+            for run_spec in run_specs:
+                scenario_spec = run_spec.scenario_spec
+                groups = run_spec.groups
+                if (
+                    scenario_spec.class_name
+                    != "helm.benchmark.scenarios.synthetic_efficiency_scenario.SyntheticEfficiencyScenario"
+                ):
+                    scenario_specs_to_groups[scenario_spec] = groups
+
+            group_to_scenario_specs: Dict = dict()
+            for scenario_spec, groups in scenario_specs_to_groups.items():
+                for group in groups:
+                    if group not in group_to_scenario_specs:
+                        group_to_scenario_specs[group] = []
+                    group_to_scenario_specs[group].append(scenario_spec)
+            return group_to_scenario_specs
+
+        def get_stats_file_metadata(data_overlap_dir: str) -> Dict[str, List[str]]:
+            """
+            Takes the data_overlap_dir as input and returns a dictionary
+            of stats_file_path -> List(model_names)
+
+            Sample input:
+            file_models_mapping:
+            - file_name: file1
+                model_names:
+                - model1
+                - model2
+            - file_name: file2
+                model_names:
+                - model2
+                - model3
+
+            """
+            metadata_file_path: str = os.path.join(data_overlap_dir, "metadata.yaml")
+            if not os.path.exists(metadata_file_path):
+                return dict()
+
+            with open(metadata_file_path, "r") as yaml_file:
+                data = yaml.safe_load(yaml_file)
+
+            if "file_models_mapping" not in data:
+                raise ValueError("Invalid YAML structure: 'file_models_mapping' key not found.")
+
+            file_metadata = {}
+            for entry in data["file_models_mapping"]:
```

Comment:
What is "this"? I'm not convinced there's much value in adding a dataclass here. For `entry`, we quickly modify it since it's relative paths to construct a separate dict. And we do not use this file/data format anywhere else so there's no interoperability concerns.

Reply:
Sure I'm happy to just remove that validation and assume the file format is structured correctly. We aren't assuming an adversarial/open setting for the file and the complexity/overhead to do more given that this file is small, used once, and hardly changes doesn't seem worth it. If things ever change, it won't be hard to add validation or a dataclass when needed